python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -226,51 +336,269 @@ def _run_iteration_impl(
+        if not group_origin:
+            return "global evaluation group"
+        else:
+            repo_name = (
+                group_origin.external_repository_origin.get_label()
+                if group_origin.external_repository_origin.repository_name
+                != SINGLETON_REPOSITORY_NAME
+                else group_origin.external_repository_origin.code_location_origin.location_name
+            )
+
+            if group_origin.instigator_name == get_default_automation_policy_sensor_name(
+                group_origin.external_repository_origin
+            ):
+                return f"default asset policy sensor in {repo_name}"
+            else:
+                return f"asset policy sensor {group_origin.instigator_name} in {repo_name}"
+
+    def _process_auto_materialize_tick(
+        self,
+        workspace_process_context: IWorkspaceProcessContext,
+        group_origin: Optional[ExternalInstigatorOrigin],
+        auto_materialize_instigator_state: Optional[InstigatorState],
+        debug_crash_flags: SingleInstigatorDebugCrashFlags,  # TODO No longer single instigator
+    ):
+        return list(
+            self._process_auto_materialize_tick_generator(
+                workspace_process_context,
+                group_origin,
+                auto_materialize_instigator_state,
+                debug_crash_flags,
+            )
+        )
+
+    def _process_auto_materialize_tick_generator(
+        self,
+        workspace_process_context: IWorkspaceProcessContext,
+        group_origin: Optional[ExternalInstigatorOrigin],
+        auto_materialize_instigator_state: Optional[InstigatorState],
+        debug_crash_flags: SingleInstigatorDebugCrashFlags,  # TODO No longer single instigator
+    ):
         evaluation_time = pendulum.now("UTC")
 
         workspace = workspace_process_context.create_request_context()
         asset_graph = ExternalAssetGraph.from_workspace(workspace)
+
+        instance: DagsterInstance = workspace_process_context.instance
+        schedule_storage = check.not_none(instance.schedule_storage)
+
+        print_group_name = self._get_display_group_name(group_origin)
+
         target_asset_keys = {
             target_key
             for target_key in asset_graph.materializable_asset_keys
             if asset_graph.get_auto_materialize_policy(target_key) is not None
+            and (
+                not group_origin
+                or get_instigator_origin_for_asset_key(target_key, asset_graph) == group_origin
+            )
         }
+
         num_target_assets = len(target_asset_keys)
 
         auto_observe_assets = [
             key
             for key in asset_graph.source_asset_keys
             if asset_graph.get_auto_observe_interval_minutes(key) is not None
+            and (
+                not group_origin
+                or get_instigator_origin_for_asset_key(key, asset_graph) == group_origin
+            )
```

Comment:
note to self - these need to be passed through to the context, it is still executing all of them